src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -64,16 +53,81 @@ private void Remove(JobDefinition? jobDefinition)
             return;
         }
 
-        allJobs.Remove(jobDefinition);
-        if (jobDefinition.CronExpression is not null)
+        JobDefinitions.Remove(jobDefinition);
+
+        // TODO: Shouldn't we also remove related entries in DependentJobsPerPrincipalJobType?
+        // cf. https://github.com/NCronJob-Dev/NCronJob/issues/107
+    }
+
+    public JobDefinition AddDynamicJob(
+        Delegate jobDelegate,
+        string? jobName = null,
+        JobOption? jobOption = null)
+    {
+        var jobPolicyMetadata = new JobExecutionAttributes(jobDelegate);
+
+        JobDefinition entry;
+
+        if (jobOption is null)
+        {
+            entry = new JobDefinition(typeof(DynamicJobFactory), null, null, null,
+                JobName: DynamicJobNameGenerator.GenerateJobName(jobDelegate),
+                JobPolicyMetadata: jobPolicyMetadata) { CustomName = jobName };
+        }
+        else
+        {
+            Debug.Assert(jobOption.CronExpression is not null);
+
+            var cron = NCronJobOptionBuilder.GetCronExpression(jobOption.CronExpression);
+
+            entry = new JobDefinition(typeof(DynamicJobFactory), null, cron, jobOption.TimeZoneInfo,
+                JobName: DynamicJobNameGenerator.GenerateJobName(jobDelegate),
+                JobPolicyMetadata: jobPolicyMetadata)
+            { CustomName = jobName, UserDefinedCronExpression = jobOption.CronExpression };
+        }
+
+        Add(entry);
+        AddDynamicJobRegistration(entry, jobDelegate);
+
+        return entry;
+    }
+
+    private void AddDynamicJobRegistration(JobDefinition jobDefinition, Delegate jobDelegate)
+        => DynamicJobRegistrations.Add(new DynamicJobRegistration(jobDefinition, sp => new DynamicJobFactory(sp, jobDelegate)));
+
+    public IJob GetDynamicJobInstance(IServiceProvider serviceProvider, JobDefinition jobDefinition)
```

Comment:
nitpick: I try to have all public methods on top and then private ones at the bottm